bundles/org.openhab.binding.vizio/src/main/java/org/openhab/binding/vizio/internal/handler/VizioHandler.java

```diff
@@ -0,0 +1,572 @@
+ */
+@NonNullByDefault
+public class VizioHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(VizioHandler.class);
+    private final HttpClient httpClient;
+    private final VizioStateDescriptionOptionProvider stateDescriptionProvider;
+    private final String dbAppsJson;
+
+    private @Nullable ServiceRegistration<?> serviceRegistration;
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable ScheduledFuture<?> metadataRefreshJob;
+
+    private VizioCommunicator communicator;
+    private List<VizioApp> userConfigApps = new ArrayList<VizioApp>();
+    private Object sequenceLock = new Object();
+
+    private int pairingDeviceId = -1;
+    private int pairingToken = -1;
+    private Long currentInputHash = 0L;
+    private Long currentVolumeHash = 0L;
+    private String currentApp = EMPTY;
+    private String currentInput = EMPTY;
+    private boolean currentMute = false;
+    private int currentVolume = -1;
+    private boolean powerOn = false;
+    private boolean debounce = true;
+
+    public VizioHandler(Thing thing, HttpClient httpClient,
+            VizioStateDescriptionOptionProvider stateDescriptionProvider, String vizioAppsJson) {
+        super(thing);
+        this.httpClient = httpClient;
+        this.stateDescriptionProvider = stateDescriptionProvider;
+        this.dbAppsJson = vizioAppsJson;
+        this.communicator = new VizioCommunicator(httpClient, EMPTY, -1, EMPTY);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Vizio handler");
+        final Gson gson = new Gson();
+        VizioConfiguration config = getConfigAs(VizioConfiguration.class);
+
+        @Nullable
+        String host = config.hostName;
+        final @Nullable String authToken = config.authToken;
+        @Nullable
+        String appListJson = config.appListJson;
+
+        if (host == null || host.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Host Name must be specified");
+            return;
+        } else if (host.contains(":")) {
+            // format for ipv6
+            host = "[" + host + "]";
+        }
+
+        this.communicator = new VizioCommunicator(httpClient, host, config.port, authToken != null ? authToken : EMPTY);
+
+        // register trustmanager service to allow httpClient to accept self signed cert from the Vizio TV
+        VizioTlsTrustManagerProvider tlsTrustManagerProvider = new VizioTlsTrustManagerProvider(
+                host + ":" + config.port);
+        serviceRegistration = FrameworkUtil.getBundle(getClass()).getBundleContext()
+                .registerService(TlsTrustManagerProvider.class.getName(), tlsTrustManagerProvider, null);
+
+        if (authToken == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                    "Auth Token must be specified, see README for instructions");
```

Comment:
Please add an i18n string for this message.

Reply:
i18 strings added for all status messages